auth-starter/src/main/java/com/vaadin/auth/starter/VaadinAuthContextImpl.java

```diff
@@ -0,0 +1,29 @@
+package com.vaadin.auth.starter;
+
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+
+import java.util.Optional;
+
+/**
+ * Default implementation of the {@link VaadinAuthContext} interface.
+ */
+
+public class VaadinAuthContextImpl implements VaadinAuthContext {
+
+    @Override
+    public Optional<OidcUser> getAuthenticatedUser() {
+        var authentication = SecurityContextHolder.getContext()
+                .getAuthentication();
```

Comment:
Not sure if that would work since we need a session-scoped bean then, and I'd avoid that if it's just to facilitate testing. BTW Spring has some security utilities for testing, such as [`@WithSecurityContext`](https://docs.spring.io/spring-security/site/docs/4.2.x/reference/html/test-method.html#test-method-withsecuritycontext) and others that can be used to easily provide mocks for tests. Did you already try using those?